services/repository/transfer.go

```diff
@@ -337,7 +337,10 @@ func changeRepositoryName(ctx context.Context, repo *repo_model.Repository, newR
 	}
 
 	if err = gitrepo.RenameRepository(ctx, repo,
-		repo_model.StorageRepo(repo_model.RelativePath(repo.OwnerName, newRepoName))); err != nil {
+		repo_model.StorageRepo{
+			RelativePath:  repo_model.RelativePath(repo.OwnerName, newRepoName),
+			DefaultBranch: repo.DefaultBranch,
```

Comment:
If `DefaultBranch` is a required field, then you can't expose it or let the callers pass it.

In the future, if someone forgets to set `DefaultBranch`, there will be a lot of new bugs.